### ISSUE TITLE
AsyncStorage extracted from react-native core

### ISSUE THREAD

**Issue by chris-feist** [rt2zz/redux-persist#1013]
I upgraded to React Native 0.59 and am now seeing this message:

`Warning: Async Storage has been extracted from react-native core and will be removed in a future release. It can now be installed and imported from '@react-native-community/async-storage' instead of 'react-native'. See https://github.com/react-native-community/react-native-async-storage`

**Comment by alexandrius** [comment]
@chris-feist I too upgraded today and redux-persist stopped working on iOS. Does it work for you?

**Comment by alexandrius** [comment]
You can use this patch to make it working again
```
diff --git a/node_modules/redux-persist/src/storage/index.native.js b/node_modules/redux-persist/src/storage/index.native.js
index 641e10f..6dcc965 100644
--- a/node_modules/redux-persist/src/storage/index.native.js
+++ b/node_modules/redux-persist/src/storage/index.native.js
@@ -1,6 +1,6 @@
 // @noflow
 /* eslint-disable */
 
-import { AsyncStorage } from 'react-native'
+import AsyncStorage from '@react-native-community/async-storage';
 
 export default AsyncStorage

```

**Comment by chris-feist** [comment]
Yeah, it still works and is just a warning message. I just posted this issue as something that will need to be resolved in the future

**Comment by alexandrius** [comment]
It fails for me when testing without debug mode or release builds

**Comment by akontaroudis** [comment]
Same as @chris-feist for me, works on debug mode on and off. I do get a warning but not a failure. As far as I can tell from react-native code this has not been removed **yet**:

https://github.com/facebook/react-native/blob/master/Libraries/react-native/react-native-implementation.js#L196

Interested to know why it's failing for you though.

**Comment by alexandrius** [comment]
Honestly I don't get why it fails for me either. It fails only on iOS

**Comment by dkpalmer** [comment]
Looks like @rt2zz has a release coming: https://github.com/rt2zz/redux-persist/releases/tag/v6.0.0-pre1

**Comment by FRizzonelli** [comment]
For me it's working like this: 
```js
import AsyncStorage from '@react-native-community/async-storage';

const persistConfig = {
  timeout: 0,
  key: 'roott',
  storage: AsyncStorage,
  blacklist: []
};
```

**Comment by stacywang0601** [comment]
When I use 

```
import AsyncStorage from '@react-native-community/async-storage';
const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
  ...
};

```

I got error saying **[@RNC/AsyncStorage]: NativeModule: AsyncStorage is null.**


Does anyone has the same issue?

**Comment by FRizzonelli** [comment]
@stacywang0601 Are you on latest RN?

**Comment by stacywang0601** [comment]
@FRizzonelli I'm on version 0.59

**Comment by FRizzonelli** [comment]
Did you run _react-native link_ after installing?

**Comment by stacywang0601** [comment]
@FRizzonelli . Seems like I didn't link it successfully.  Thanks!

**Comment by MiltonBO** [comment]
> When I use
> 
> ```
> import AsyncStorage from '@react-native-community/async-storage';
> const persistConfig = {
>   key: 'root',
>   storage: AsyncStorage,
>   ...
> };
> ```
> 
> I got error saying **[@RNC/AsyncStorage]: NativeModule: AsyncStorage is null.**
> 
> Does anyone has the same issue?

Check you have installed the async storage with npm, restart your npm server and restart your emulator.

**Comment by varun761** [comment]
Hello, i am getting the same error does anyone find the solution with expo?

**Comment by MiltonBO** [comment]
Yes, this is the solution:

https://github.com/rt2zz/redux-persist/issues/1013#issuecomment-513586270

**Comment by varun761** [comment]
@MiltonBO I used that . Didn't works .

**Comment by MiltonBO** [comment]
Every library that use AsyncStorage must be changed to import from community library. When you let a library to import from 'react-native' that warning will happens.

I found a lot of libraries using AsyncStorage, check what library is having the problem. If you changed this for redux-persist, could be another library the problem.

Don't forget to restart your metro server and reinstall the mobile app in your device.

**Comment by HugoLiconV** [comment]
> When I use
> 
> ```
> import AsyncStorage from '@react-native-community/async-storage';
> const persistConfig = {
>   key: 'root',
>   storage: AsyncStorage,
>   ...
> };
> ```
> 
> I got error saying **[@RNC/AsyncStorage]: NativeModule: AsyncStorage is null.**
> 
> Does anyone has the same issue?

I'm having the same error, I'm on version 0.59, I already:
- run `react-native link @react-native-community/async-storage`
- run `react-native run-android`
- restart emulator

**Comment by ramyareye** [comment]
@HugoLiconV `pod install` maybe ?

**Comment by HugoLiconV** [comment]
> @HugoLiconV `pod install` maybe ?

The problem was that it wasn't linking correctly, so I had to do it manually

**Comment by paulmars** [comment]
i was using Async for session management and the Async removeItem was reporting success but failing to delete the item (user stayed logged in). When I removed redux-persist the message disappeared (obviously) and Async started working again.

Another comment suggested that you can't include both (one from library and one from a custom view) and expect consistent results.

**Comment by jbrodriguez** [comment]
I'm having this issue on iOS ... it doesn't. persist the state with AsyncStorage.

If I call @RNC/AsyncStorage independently from redux-persist, the data is actually stored.
